app/src/org/commcare/CommCareApplication.java

```diff
@@ -982,17 +977,24 @@ public void clearPendingUserMessage() {
     }
 
     private static void setupLoggerStorage(boolean userStorageAvailable) {
-        CommCareApplication app = CommCareApplication.instance();
+        boolean loggingEnabled = true;
+        if (instance().currentApp != null) {

```

Comment:
It would be cleaner to do this check inside the `isLoggingEnabled()` method and return true there if there's no current app